sqlmesh/core/model/meta.py

```diff
@@ -238,8 +238,9 @@ def _kind_validator(cls, values: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
             if not kind.is_materialized:
                 if values.get("partitioned_by_"):
                     raise ValueError(f"partitioned_by field cannot be set for {kind} models")
-            if values.get("batch_size") and not kind.supports_batch_size:
-                raise ValueError(f"batch_size field cannot be set for {kind} models")
+            for prop in ("batch_size", "lookback"):
+                if values.get(prop) and not kind.is_incremental:
+                    raise ValueError(f"{prop} field cannot be set for {kind} models")
```

Comment:
Maybe provide more context. `{prop} field only works with incremental models, cannot be set for {kind} models`